src/trace/ord/key_batch.rs

```diff
@@ -329,11 +329,13 @@ where
 
     fn map_times_through<L: FnMut(&T, &R)>(&mut self, mut logic: L, upper: &T) {
         self.cursor.child.rewind();
-        while self.cursor.child.valid() && self.cursor.child.key().0.less_equal(upper) {
-            logic(
-                self.cursor.child.current_key(),
-                self.cursor.child.current_diff(),
-            );
+        while self.cursor.child.valid() {
+            if self.cursor.child.key().0.less_equal(upper) {
+                logic(
+                    self.cursor.child.current_key(),
+                    self.cursor.child.current_diff(),
+                );
+            }
```

Comment:
We probably need some tests covering this

Reply:
We do now, in aggregation operator test harness.  We do need dedicated unit tests for cursors though.

Reply:
Do we? I mentioned it because if the coverage warning saying the function was uncovered

Reply:
Ah, I guess that one tests `val_batch`